### PR TITLE
Add identification headers to SDK communication with servers

### LINKED ISSUE #739: Identifying headers missing from SDK requests
The 'User Agent' and 'X-Planet-App' headers are used to self-identify the SDK when communicating with the Planet services. These headers are not being sent. Fix the SDK logic so they are indeed being fixed.

example:
```console
$> planet orders request --id 20221103_091114_08_2451 --name test PSScene visual | planet --verbosity debug orders create

2022-11-03 14:02:18,244 - asyncio - DEBUG - Using selector: KqueueSelector
2022-11-03 14:02:18,244 - planet.auth - DEBUG - Reading from /Users/jennifer.kyle/.planet.json
2022-11-03 14:02:18,245 - planet.auth - DEBUG - Auth read from secret file /Users/jennifer.kyle/.planet.json.
2022-11-03 14:02:18,245 - planet.http - INFO - Session read timeout set to 30.0.
2022-11-03 14:02:18,257 - planet.http - DEBUG - Throttling cadence set to 0.1s.
2022-11-03 14:02:18,257 - planet.http - DEBUG - Workers capped at 50.
2022-11-03 14:02:18,258 - planet.http - DEBUG - Worker acquired.
2022-11-03 14:02:18,258 - planet.http - INFO - POST https://api.planet.com/compute/ops/orders/v2 - Sent
2022-11-03 14:02:18,258 - planet.http - DEBUG - POST https://api.planet.com/compute/ops/orders/v2 - Headers({'host': 'api.planet.com', 'content-type': 'application/json', 'content-length': '151', 'authorization': '[secure]'})
2022-11-03 14:02:18,258 - planet.http - DEBUG - POST https://api.planet.com/compute/ops/orders/v2 - b'{"name": "test", "products": [{"item_ids": ["20221103_091114_08_2451"], "item_type": "PSScene", "product_bundle": "visual"}], "metadata": {"stac": {}}}'
2022-11-03 14:02:19,047 - httpx._client - DEBUG - HTTP Request: POST https://api.planet.com/compute/ops/orders/v2 "HTTP/1.1 202 Accepted"
2022-11-03 14:02:19,047 - planet.http - INFO - POST https://api.planet.com/compute/ops/orders/v2 - Status 202
2022-11-03 14:02:19,047 - planet.http - DEBUG - Headers({'content-type': 'application/json', 'date': 'Thu, 03 Nov 2022 21:02:19 GMT', 'strict-transport-security': 'max-age=15724800; includeSubDomains', 'vary': 'Origin', 'x-planet-span-id': '9fefdcee-659c-4b03-9373-984a612aa32f', 'x-planet-trace-id': 'e4985228-f779-47ae-bb7e-b69d92e52576', 'x-request-id': 'e4985228-f779-47ae-bb7e-b69d92e52576', 'content-length': '438', 'via': '1.1 edge, 1.1 google', 'alt-svc': 'h3=":443"; ma=2592000,h3-29=":443"; ma=2592000'})
2022-11-03 14:02:19,048 - planet.http - DEBUG - Worker released.
{"_links": {"_self": "https://api.planet.com/compute/ops/orders/v2/bfcc1c34-8b3b-4e91-addf-dfef8fc90b5f"}, "created_on": "2022-11-03T21:02:18.673Z", "error_hints": [], "id": "bfcc1c34-8b3b-4e91-addf-dfef8fc90b5f", "last_message": "Preparing order", "last_modified": "2022-11-03T21:02:18.673Z", "metadata": {"stac": {}}, "name": "test", "products": [{"item_ids": ["20221103_091114_08_2451"], "item_type": "PSScene", "product_bundle": "visual"}], "state": "queued"}
```
In the output, the following line should have the 'User Agent' and 'X-Planet-App' headers but they are missing:
```
2022-11-03 14:02:18,258 - planet.http - DEBUG - POST https://api.planet.com/compute/ops/orders/v2 - Headers({'host': 'api.planet.com', 'content-type': 'application/json', 'content-length': '151', 'authorization': '[secure]'})
```
The session-level headers are getting set when the Session is created, but they are not merged into the request-level headers when a request is made. This merging happens in httpx with [build_request()](https://github.com/encode/httpx/blob/1aea9539bbe93b26103e3a722ba0c421f7eb7f82/httpx/_client.py#L322), but we are bypassing that function and going directly to calling send(). So we do not get the merging behavior httpx provides.

Two options:
1. move up a level from calling send() directly so as to use httpx's merging logic *(may take some work figuring out why we went directly to send())*
1. merge headers manually in Session._request() *(easier but wanders farther off the usage path httpx anticipates)*

### PR DESCRIPTION
**Pull request by jreiberkyle**
**Related Issue(s):**

Closes #739 

**Proposed Changes:**

For inclusion in changelog:
1. Client identification headers are now sent with all requests.

Not intended for changelog:

1. The Request model has been removed and Sessions.request() and Sessions.stream() now take in arguments such as url, method, json, etc, and create the request internally.
1. All clients now have their methods call Session.request() directly instead of all going through a private method of the class
1. The Paged class in models now takes in the first page response as an argument instead of starting with a Request object and submitting the request and handling the first page internally. This was necessitated by the removal of the Request object but also has the benefit of earlier notification to the user of an error when the first page request was unsuccessful. Previously, the first page was only requested (and error thrown) when the user started using the generator.
1. Adds tests for confirming that the headers are sent in requests and a test for orders client to confirm the expected request is sent to the server.

**Diff of User Interface**

Old behavior:

From the code and output below, this is the relevant line:
```
2022-11-09 20:07:38,294 - planet.http - DEBUG - POST https://api.planet.com/compute/ops/orders/v2 - Headers({'host': 'api.planet.com', 'content-type': 'application/json', 'content-length': '158', 'authorization': '[secure]'})
```
Note that the headers 'x-planet-app' and 'user-agent' are not included.

```console
planet-client-python$> planet orders request --name test --id 20221109_052227_10_2432 PSScene analytic_udm2 | planet --verbosity debug orders create
2022-11-09 20:07:38,281 - asyncio - DEBUG - Using selector: KqueueSelector
2022-11-09 20:07:38,282 - planet.auth - DEBUG - Reading from /Users/jennifer.kyle/.planet.json
2022-11-09 20:07:38,282 - planet.auth - DEBUG - Auth read from secret file /Users/jennifer.kyle/.planet.json.
2022-11-09 20:07:38,282 - planet.http - INFO - Session read timeout set to 30.0.
2022-11-09 20:07:38,294 - planet.http - DEBUG - Throttling cadence set to 0.1s.
2022-11-09 20:07:38,294 - planet.http - DEBUG - Workers capped at 50.
2022-11-09 20:07:38,294 - planet.http - DEBUG - Worker acquired.
2022-11-09 20:07:38,294 - planet.http - INFO - POST https://api.planet.com/compute/ops/orders/v2 - Sent
2022-11-09 20:07:38,294 - planet.http - DEBUG - POST https://api.planet.com/compute/ops/orders/v2 - Headers({'host': 'api.planet.com', 'content-type': 'application/json', 'content-length': '158', 'authorization': '[secure]'})
2022-11-09 20:07:38,295 - planet.http - DEBUG - POST https://api.planet.com/compute/ops/orders/v2 - b'{"name": "test", "products": [{"item_ids": ["20221109_052227_10_2432"], "item_type": "PSScene", "product_bundle": "analytic_udm2"}], "metadata": {"stac": {}}}'
2022-11-09 20:07:39,033 - httpx._client - DEBUG - HTTP Request: POST https://api.planet.com/compute/ops/orders/v2 "HTTP/1.1 202 Accepted"
2022-11-09 20:07:39,033 - planet.http - INFO - POST https://api.planet.com/compute/ops/orders/v2 - Status 202
2022-11-09 20:07:39,033 - planet.http - DEBUG - Headers({'content-type': 'application/json', 'date': 'Thu, 10 Nov 2022 04:07:39 GMT', 'strict-transport-security': 'max-age=15724800; includeSubDomains', 'vary': 'Origin', 'x-planet-span-id': '3178ea59-5cb8-444e-a272-ce0812008a28', 'x-planet-trace-id': '369209dd-9d29-43ae-8d98-a0281e552530', 'x-request-id': '369209dd-9d29-43ae-8d98-a0281e552530', 'content-length': '445', 'via': '1.1 edge, 1.1 google', 'alt-svc': 'h3=":443"; ma=2592000,h3-29=":443"; ma=2592000'})
2022-11-09 20:07:39,033 - planet.http - DEBUG - Worker released.
{"_links": {"_self": "https://api.planet.com/compute/ops/orders/v2/ce65731e-ee90-4dd4-8950-29ba5496b47f"}, "created_on": "2022-11-10T04:07:38.742Z", "error_hints": [], "id": "ce65731e-ee90-4dd4-8950-29ba5496b47f", "last_message": "Preparing order", "last_modified": "2022-11-10T04:07:38.742Z", "metadata": {"stac": {}}, "name": "test", "products": [{"item_ids": ["20221109_052227_10_2432"], "item_type": "PSScene", "product_bundle": "analytic_udm2"}], "state": "queued"}
```

New behavior:

From the code and output below, this is the relevant line:
```
2022-11-09 20:08:58,295 - planet.http - DEBUG - POST https://api.planet.com/compute/ops/orders/v2 - Headers({'host': 'api.planet.com', 'accept': '*/*', 'accept-encoding': 'gzip, deflate', 'connection': 'keep-alive', 'user-agent': 'planet-client-python/2.0a6dev', 'x-planet-app': 'python-cli', 'content-type': 'application/json', 'content-length': '158', 'authorization': '[secure]'})
```
Note that the headers 'x-planet-app' and 'user-agent' are now included.

```console
planet-client-python$> planet orders request --name test --id 20221109_052227_10_2432 PSScene analytic_udm2 | planet --verbosity debug orders create
2022-11-09 20:08:58,282 - asyncio - DEBUG - Using selector: KqueueSelector
2022-11-09 20:08:58,282 - planet.auth - DEBUG - Reading from /Users/jennifer.kyle/.planet.json
2022-11-09 20:08:58,282 - planet.auth - DEBUG - Auth read from secret file /Users/jennifer.kyle/.planet.json.
2022-11-09 20:08:58,282 - planet.http - INFO - Session read timeout set to 30.0.
2022-11-09 20:08:58,294 - planet.http - DEBUG - Throttling cadence set to 0.1s.
2022-11-09 20:08:58,294 - planet.http - DEBUG - Workers capped at 50.
2022-11-09 20:08:58,295 - planet.http - DEBUG - Worker acquired.
2022-11-09 20:08:58,295 - planet.http - INFO - POST https://api.planet.com/compute/ops/orders/v2 - Sent
2022-11-09 20:08:58,295 - planet.http - DEBUG - POST https://api.planet.com/compute/ops/orders/v2 - Headers({'host': 'api.planet.com', 'accept': '*/*', 'accept-encoding': 'gzip, deflate', 'connection': 'keep-alive', 'user-agent': 'planet-client-python/2.0a6dev', 'x-planet-app': 'python-cli', 'content-type': 'application/json', 'content-length': '158', 'authorization': '[secure]'})
2022-11-09 20:08:58,295 - planet.http - DEBUG - POST https://api.planet.com/compute/ops/orders/v2 - b'{"name": "test", "products": [{"item_ids": ["20221109_052227_10_2432"], "item_type": "PSScene", "product_bundle": "analytic_udm2"}], "metadata": {"stac": {}}}'
2022-11-09 20:08:58,986 - httpx._client - DEBUG - HTTP Request: POST https://api.planet.com/compute/ops/orders/v2 "HTTP/1.1 202 Accepted"
2022-11-09 20:08:58,986 - planet.http - INFO - POST https://api.planet.com/compute/ops/orders/v2 - Status 202
2022-11-09 20:08:58,986 - planet.http - DEBUG - Headers({'content-encoding': 'gzip', 'content-type': 'application/json', 'date': 'Thu, 10 Nov 2022 04:08:59 GMT', 'strict-transport-security': 'max-age=15724800; includeSubDomains', 'vary': 'Origin', 'x-planet-span-id': 'b8141866-f147-4e5c-867f-928c87e7f3c9', 'x-planet-trace-id': '8f59f424-7a2d-4dfa-aeed-281c1cb5dd6f', 'x-request-id': '8f59f424-7a2d-4dfa-aeed-281c1cb5dd6f', 'via': '1.1 edge, 1.1 google', 'alt-svc': 'h3=":443"; ma=2592000,h3-29=":443"; ma=2592000', 'transfer-encoding': 'chunked'})
2022-11-09 20:08:58,988 - planet.http - DEBUG - Worker released.
{"_links": {"_self": "https://api.planet.com/compute/ops/orders/v2/93fb9b71-cbd7-4ed7-be8b-c04c0fc08ec2"}, "created_on": "2022-11-10T04:08:58.733Z", "error_hints": [], "id": "93fb9b71-cbd7-4ed7-be8b-c04c0fc08ec2", "last_message": "Preparing order", "last_modified": "2022-11-10T04:08:58.733Z", "metadata": {"stac": {}}, "name": "test", "products": [{"item_ids": ["20221109_052227_10_2432"], "item_type": "PSScene", "product_bundle": "analytic_udm2"}], "state": "queued"}
```

**PR Checklist:**

- [x] This PR is as small and focused as possible
- [x] The title of this PR and/or the list of proposed changes are ready for inclusion in the Changelog or I have determined a Changelog entry is not required
- [x] I have updated docstrings for function changes and docs in the 'docs' folder for user interface / behavior changes
- [x] This PR does not break any examples or I have updated them

**(Optional) @mentions for Notifications:**
@cholmes @sgillies

### CODE REVIEW

File: planet/clients/data.py [hunk 1]
```diff
@@ -260,15 +250,16 @@ async def update_search(self,
...
             'filter': search_filter,
             'item_types': item_types,
             '__daily_email_enabled': enable_email
         }
 
-        request = self._request(url, method='PUT', json=request_json)
-        response = await self._do_request(request)
+        response = await self._session.request(method='PUT',
+                                               url=url,
+                                               json=request)
```
Comment (sgillies) [inline planet/clients/data.py:262]: @jreiberkyle @kevinlacaille tangential comment here: the code might be more readable if we didn't use "request" as both a verb and a noun. Maybe `self._session.request` could be changed to `self._session.send` someday?

File: planet/clients/subscriptions.py [hunk 2]
```diff
@@ -63,11 +63,13 @@ class _SubscriptionsPager(Paged):
...
-        req = Request(url, params=params)
 
         try:
-            async for sub in _SubscriptionsPager(req,
-                                                 self.session.request,
+            response = await self._session.request(method='GET',
+                                                   url=url,
+                                                   params=params)
+            async for sub in _SubscriptionsPager(response,
+                                                 self._session.request,
```
Comment (sgillies) [inline planet/clients/subscriptions.py:72]: Great.

File: tests/unit/test_models.py [hunk 3]
```diff
@@ -168,111 +149,45 @@ async def _aiter_bytes():
...
+    resp.json = lambda: {'_links': {'next': 'blah'}, 'items': [1, 2]}
 
-    async def func(req):
-        return mock_http_response(json={
-            '_links': {
-                'next': 'blah'
-            }, 'items': [1, 2]
-        })
+    async def get_response(url, method):
+        return resp
```
Comment (sgillies) [inline tests/unit/test_models.py:188]: I like that the pager change makes testing more clear.

**Review by sgillies — APPROVED** [review]
@jreiberkyle excellent :rocket:

File: planet/clients/data.py (hunk 1 above)
Comment (jreiberkyle) [reply planet/clients/data.py:262]: Ohh yeah I fully agree here! Right now Session has two methods that communicate with the servers: request() and stream(). Both of these suffer from verb/noun ambiguity. This was some of the satisfaction in removing the Request and Stream classes. At least we no longer have noun representation in the code base.

**Pull request merged**